awx_collection/plugins/modules/tower_inventory_source.py

```diff
@@ -190,23 +196,30 @@ def main():
     name = module.params.get('name')
     new_name = module.params.get('new_name')
     inventory = module.params.get('inventory')
+    organization = module.params.get('organization')
     source_script = module.params.get('source_script')
     credential = module.params.get('credential')
     source_project = module.params.get('source_project')
     state = module.params.get('state')
 
-    # Attempt to look up inventory source based on the provided name and inventory ID
-    inventory_id = module.resolve_name_to_id('inventories', inventory)
-    inventory_source = module.get_one('inventory_sources', **{
+    lookup_data = {'name': inventory}
+    if organization:
+        lookup_data['organization'] = module.resolve_name_to_id('organizations', organization)
+    inventory_object = module.get_one('inventories', data=lookup_data)
+
+    if not inventory_object:
+        module.fail_json(msg='The specified inventory was not found.')
```

Comment:
Non-blocking comment, but I think it would be helpful to put `lookup_data` in this message with a `.format`.